src/Build/BackEnd/Components/Communications/SerializationContractInitializer.cs

```diff
@@ -30,7 +29,10 @@ private static void RegisterExceptions()
                 new(typeof(InternalLoggerException), (msg, inner) => new InternalLoggerException(msg, inner)),
                 new(typeof(InvalidProjectFileException), (msg, inner) => new InvalidProjectFileException(msg, inner)),
                 new(typeof(InvalidToolsetDefinitionException), (msg, inner) => new InvalidToolsetDefinitionException(msg, inner)),
-                new(typeof(ProjectCacheException), (msg, inner) => new ProjectCacheException(msg, inner)),
+                new(typeof(ProjectCache.ProjectCacheException), (msg, inner) => new ProjectCache.ProjectCacheException(msg, inner)),
+#pragma warning disable CS0618 // Type or member is obsolete
+                new(typeof(Experimental.ProjectCache.ProjectCacheException), (msg, inner) => new ProjectCache.ProjectCacheException(msg, inner)),
```

Comment:
shouldn't you instantiate **Experimental**.ProjectCache.ProjectCacheException here?

Reply:
This is intentional, an experimental plugin can throw the experimental exception but I don't want to duplicate handler code. So convert it to the new type right away.

Reply:
I agree, but worth the comment please.